klarna-on-demand/src/main/java/com/klarna/ondemand/RegistrationActivity.java

```diff
@@ -19,10 +22,46 @@
      */
     private static final String PAYLOAD_USER_TOKEN = "userToken";
     public static final String EXTRA_USER_TOKEN = "userToken";
+    private static String myProfilePhoneNumber = null;
+
+    public static final String EXTRA_SETTINGS = "settings";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        initializeLoader();
+    }
+
+    void initializeLoader() {
+        getLoaderManager().initLoader(0, null, this);
+    }
 
     @Override
     protected String getUrl() {
-        return UrlHelper.registrationUrl(getApplicationContext());
+        RegistrationSettings settings = (RegistrationSettings) getIntent().getSerializableExtra(EXTRA_SETTINGS);
+
+        if(settings == null) {
+            settings = new RegistrationSettingsBuilder().build();
```

Comment:
Correct me if I'm wrong, but isn't this the only time we use the `RegistrationSettingsBuilder` to build settings? Seeing how they are blank, I again question its existence.